src/test/java/app/daos/MovieDAOTest.java

```diff
@@ -1,117 +1,118 @@
-package org.example.daos;
+package app.daos;
 
 import jakarta.persistence.EntityManager;
 import jakarta.persistence.EntityManagerFactory;
 import jakarta.persistence.EntityNotFoundException;
-import org.example.config.HibernateConfig;
-import org.example.dtos.CastMemberDTO;
-import org.example.dtos.GenreDTO;
-import org.example.dtos.MovieDTO;
-import org.example.entities.Genre;
-import org.example.entities.Movie;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import app.config.HibernateConfig;
+import app.dtos.CastMemberDTO;
+import app.dtos.GenreDTO;
+import app.dtos.MovieDTO;
+import app.entities.Genre;
+import app.entities.Movie;
+import org.junit.jupiter.api.*;
 
 import java.time.LocalDate;
-import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 class MovieDAOTest {
 
-    private static EntityManagerFactory emf;
+    private static EntityManagerFactory emfTest;
     private static MovieDAO movieDAO;
 
-    private List<GenreDTO> genreDTOS;
-    private List<MovieDTO> movieDTOS;
+    private List<GenreDTO> genreDTOs;
+    private List<MovieDTO> movieDTOs;
     private List<CastMemberDTO> castMemberDTOs;
 
     @BeforeAll
     static void beforeAll() {
-        emf = HibernateConfig.getEntityManagerFactoryForTest();
-        movieDAO = new MovieDAO(emf);
+        emfTest = HibernateConfig.getEntityManagerFactoryForTest();
+        movieDAO = new MovieDAO(emfTest);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        emfTest.close();
```

Comment:
Remove tearDown closing the entity manager factory. The factory is a singleton within HibernateConfig so closing it will prevent other test classes from running. The program terminates after tests anyway so it's probably not necessary to close it

Reply:
Agree that a teardown i not entirely necessary in our case since we make use of try-with-resources in setup. But closing it should not have an effect on other test classes since they do get created again in each test class and the idea was to make sure that, when working on same database, that there's isolation between each test class and we start from a clean slate. But it is an expensive operation between each class and should technically only be created and closed before and after the entire test suite.

Reply:
When calling getEntityManagerFactoryForTest and the entity manager happened to be closed in one test class it will then create a new EntityManagerFactory when we call getEntityManagerFactoryForTest in another test class as seen in their HibernateConfig class...